app/models/author.rb

```diff
@@ -1,6 +1,8 @@
 class Author < ApplicationRecord
-  has_many :books
-  has_many :published, foreign_key: :publisher_id, class_name: 'Book', as: :publisher
+  has_many :books, dependent: :destroy
+  has_many :published, foreign_key: :publisher_id, class_name: 'Book', as: :publisher, dependent: :destroy, inverse_of: :author
+
+  validates :name, presence: true
```

Comment:
I felt like adding at least one validation for the author as it does not make any sense i would have an author object without his name